avocado/plugins/runner_nrunner.py

```diff
@@ -16,18 +16,96 @@
 NRunner based implementation of job compliant runner
 """
 
+import asyncio
 import json
+import multiprocessing
 import os
-import time
+import random
 from copy import copy
 
 from avocado.core import nrunner
+from avocado.core.dispatcher import SpawnerDispatcher
+from avocado.core.plugin_interfaces import CLI, Init
 from avocado.core.plugin_interfaces import Runner as RunnerInterface
+from avocado.core.settings import settings
 from avocado.core.status.repo import StatusRepo
+from avocado.core.status.server import StatusServer
 from avocado.core.task.runtime import RuntimeTask
+from avocado.core.task.statemachine import TaskStateMachine, Worker
 from avocado.core.test_id import TestID
 
 
+class RunnerInit(Init):
+
+    name = 'nrunner'
+    description = '*EXPERIMENTAL* nrunner initialization'
+
+    def initialize(self):
+        section = 'nrunner'
+        help_msg = 'Shuffle the tasks to be executed'
```

Comment:
What is a standard use case for shuffling?

Reply:
I've found that when you have a mix of tests that are short (and thus CPU bound) and longer tests (usually I/O bound) you can be wasting resources if you do:

```
avocado run --test-runner=nrunner cpu/bound/tests/* i/o/bound/tests*
```

If you mix them, the overall job can be quite faster.  In the future, we want the "scheduler" to be able to pick a better mix of tests according to policies/profiles.  Such as a "mix CPU and I/O bound tests" policy.

Reply:
Yes, perhaps the future scheduler could also measure test durations and store them in some reusable format that could then be parsed on later runs and used for most optimal resource assignment. This at least on the pure CPU-side though. There could be many policies that would be better than a randomized baseline policy.

Reply:
> Yes, perhaps the future scheduler could also measure test durations and store them in some reusable format that could then be parsed on later runs and used for most optimal resource assignment. This at least on the pure CPU-side though. There could be many policies that would be better than a randomized baseline policy.

Are you reading my mind??? :)